default.nix

```diff
@@ -0,0 +1,84 @@
+# Building purebred
+#
+# You should be able to simply invoke:
+#
+# $ nix-build
+#
+# or, to be explicit:
+#
+# $ nix-build default.nix
+#
+# in purebred's root directory in order to build purebred. You'll find the binary under:
+#
+# $ ls result/bin/purebred
+#
+# if the build was successful.
+#
+#
+# Choosing a different compiler than the default
+#
+# In order to choose a different compiler, invoke nix build like so (escaping
+# the quotes is needed, since we're passing a string literal):
+#
+# $ nix-build --arg compiler \"ghc442\"
+#
+#
+# Use as a development environment
+#
+# $ nix-shell default.nix
+#
+{ compiler ? "ghc822" }:
+
+let
+  config = {
+    packageOverrides = super: let self = super.pkgs; in
+    {
+      haskell = super.haskell // {
+        packages = super.haskell.packages // {
+          "${compiler}" = super.haskell.packages."${compiler}".override {
+            overrides = self: super: {
+              # Build with latest known stable version.
+              purebred = with pkgs.haskell.lib; dontHaddock (self.callPackage ./.nix-helpers/purebred.nix { });
+              purebred-email = self.callPackage ./.nix-helpers/purebred-email.nix { };
+              # Version pins since those versions were bleeding edge and not on
+              # hackage/stackage or Nix yet. Remove these once builds without
+              # these pass.
+              notmuch = self.callPackage ./.nix-helpers/hs-notmuch.nix {
+                notmuch = pkgs.notmuch;
+                talloc = pkgs.talloc;
+              };
+              brick = self.callPackage ./.nix-helpers/brick.nix { };
+              vty = self.callPackage ./.nix-helpers/vty.nix { };
+              Cabal = self.callPackage ./.nix-helpers/Cabal.nix { };
+            };
+          };
+        };
+      };
+    };
+  };
+  pkgs = import <nixpkgs> { inherit config; };
+  env = pkgs.haskell.packages.${compiler}.ghcWithPackages (self: [
+    self.purebred
+  ]);
+  in
+    if pkgs.lib.inNixShell
+    then pkgs.lib.overrideDerivation pkgs.haskell.packages.${compiler}.purebred.env (old: {
+         buildInputs = old.buildInputs ++ [
```

Comment:
Can we clean this up?  Is it necessary?

Reply:
No technical reason. Just took it out to save a few seconds built time. Will remove it.

Reply:
Ignore the previous comment. Do you mean the commented out line? Yeah I thought I'd keep it as an example, but yeah - happy to remove it too.

Reply:
Done.